transport/http/outbound.go

```diff
@@ -51,6 +51,14 @@ type outbound struct {
 }
 
 func (o outbound) Call(ctx context.Context, req *transport.Request) (*transport.Response, error) {
+	start := time.Now()
+	if req.TTL != 0 {
+		var cancel func()
+		ctx, cancel = context.WithDeadline(ctx, start.Add(req.TTL))
```

Comment:
optional: Instead of `WithDeadline`, you can use `WithTimeout` here and not have to add the `start + TTL`)